src/main/java/com/team4/testingsystem/services/impl/ContentFilesServiceImpl.java

```diff
@@ -31,10 +40,26 @@ public ContentFile getById(long id) {
     }
 
     @Override
-    public ContentFile add(String url, List<Question> questions) {
+    public ContentFile add(Resource file, List<Question> questions) {
+        String url = storageService.upload(file);
         return contentFilesRepository.save(new ContentFile(url, questions));
     }
 
+
+    @Override
+    public ContentFile update(MultipartFile file, Long id, List<Question> questions) {
+        if (file == null) {
+            questionService.archiveQuestionsByContentFileId(id);
+            ContentFile contentFile = contentFilesRepository
+                    .findById(id).orElseThrow(FileNotFoundException::new);
+            contentFile.getQuestions().addAll(questions);
+            return contentFilesRepository.save(contentFile);
```

Comment:
Is it possible, that between `archiveQuestions` and save `ContentFile` with a new one someone will generate a test with it, or open the content file to edit?
If so, I think it's inappropriate and we should deal with it. I could suggest you 2 options:
1.  make the method `@Transactional`. If the changes will be made in a single transaction, we shouldn't have any problems with it
2. Even though we are not updating the file, we can create a new `ContentFile` with the same URL data and archive the previous one.